### DISCUSSION TITLE
Holes in my stats.

### DISCUSSION THREAD

**Discussion by jonathh21** [BottlecapDave/HomeAssistant-OctopusEnergy#542] [Q&A] [answered]
I have holes in my stats. For the most part I remember a blip or something on that day and it doesn't seem to try again.

In there a way to re-trigger certain days or rebase the whole lot.

Thanks!

**Comment by BottlecapDave** [comment, marked as answer]
Hello - there's a service available as per the docs https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/services.md#service-octopus_energyrefresh_previous_consumption_data

**Comment by jonathh21** [comment]
nice one - I thought i'd seen it somewhere.

You call out it is only good for thethose 2 comsumption entities - what about the cost ones that back it? can you refresh the cost too?

Thank you.

**Reply by jonathh21** [reply]
So the electric one ran sweet as a nut and updated.

the gas one is giving me jip - m3 ran without error - but the kwh one errors and that is the one that is used..

any tips?

**Reply by BottlecapDave** [reply]
As per the docs, the kWh one isn't supported, which was an overlook on my side. I'll try and add support in a future release, but it'll miss the next release.

**Reply by jonathh21** [reply]
My bad - missed that wasn't supported.

Do the costs need to be refreshed as well or is that covered by refreshing the usage?

**Reply by BottlecapDave** [reply]
Cost should be refreshed at the same time as should peak/off peak breakdowns

**Reply by BottlecapDave** [reply]
v9.0.1 should now fill in the missing kwh rates if you re-run the service against `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_previous_accumulative_consumption`
